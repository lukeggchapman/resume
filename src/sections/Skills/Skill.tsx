import React from 'react'
import { Text, View, StyleSheet } from '@react-pdf/renderer'

import typography from '../../components/typography'

const styles = StyleSheet.create({
  container: {
    marginBottom: 10,
  },
  title: {
    textTransform: 'capitalize',
    marginBottom: 5,
  },
  skill: {
    color: '#333',
  },
})

export interface SkillProps {
  title: string
  skills: string[]
}

const Skill = ({ title, skills }: SkillProps) => (
  <View style={styles.container}>
    <Text style={[typography.h3, styles.title]}>{title}</Text>
    {skills.map(skill => (
      <Text key={skill} style={[typography.h4, styles.skill]}>
        {skill}
      </Text>
    ))}
  </View>
)

export default Skill
