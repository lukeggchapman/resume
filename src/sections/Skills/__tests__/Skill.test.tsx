import React from 'react'
import { shallow } from 'enzyme'

import Skill, { SkillProps } from '../Skill'

const baseProps: SkillProps = {
  title: 'titleTest',
  skills: ['skillTest1', 'skillTest2'],
}

describe('Skill', () => {
  const render = (props?: Partial<SkillProps>) =>
    shallow(<Skill {...{ ...baseProps, ...props }} />)

  it('renders correctly', () => {
    expect(render()).toMatchSnapshot()
  })
})
