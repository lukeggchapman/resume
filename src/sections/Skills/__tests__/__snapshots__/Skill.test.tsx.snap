// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Skill renders correctly 1`] = `
<View
  style={
    {
      "marginBottom": 10,
    }
  }
>
  <Text
    style={
      [
        {
          "fontFamily": "Oswald",
          "fontSize": 14,
        },
        {
          "marginBottom": 5,
          "textTransform": "capitalize",
        },
      ]
    }
  >
    titleTest
  </Text>
  <Text
    key="skillTest1"
    style={
      [
        {
          "color": "#555",
          "fontFamily": "PT Sans",
          "fontSize": 10,
        },
        {
          "color": "#333",
        },
      ]
    }
  >
    skillTest1
  </Text>
  <Text
    key="skillTest2"
    style={
      [
        {
          "color": "#555",
          "fontFamily": "PT Sans",
          "fontSize": 10,
        },
        {
          "color": "#333",
        },
      ]
    }
  >
    skillTest2
  </Text>
</View>
`;
